mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -130,6 +130,25 @@ class GaussianMixtureModel private[ml] (
   }
 
   /**
+   * Return the total log-likelihood for this model on the given data.
+   */
+  @Since("2.2.0")
+  def computeLogLikelihood(dataset: Dataset[_]): Double = {
```

Comment:
+1 for putting it in the summary.  If you want to evaluate a new dataset, then let's add an evaluate() method which returns a summary.